util/ci.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+if [ -z "$TEST_TOOL" ]
+then
+  echo "You must specify a TEST_TOOL"
+  exit 1
+fi
+
+case $1 in
+  before_script)
+    if [ "$TEST_TOOL" = "rubygems" ]
+    then
+      gemhome=$(gem env home)
+
+      ruby -I lib bin/gem uninstall executable-hooks gem-wrappers bundler-unload -x --force -i "$gemhome@global"
+
+      rake setup
+
+      gem list --details
+      gem env
+    else
+      cd bundler
+
+      if [ -n "$BDV" ]
+      then
+        git reset --hard "origin/$BDV"
+      fi
+
+      rake spec:travis:deps
+    fi
+
+    ;;
+
+  rubocop)
+    gem install rubocop -v "~>0.71.0"
+    util/rubocop
+
+    ;;
+
+  script)
+    if [ "$TEST_TOOL" = "rubygems" ]
+    then
+      rake test
```

Comment:
Prefixing with `exec ` here and below could be useful to avoid keeping an intermediate Bash process.
It would also ensures that if `rake` exits with a non-zero code, that it is correctly propagated.